Sources/MapboxCoreNavigation/SimulatedLocationManager.swift

```diff
@@ -160,9 +162,9 @@ open class SimulatedLocationManager: NavigationLocationManager {
                                   course: newCoordinate.direction(to: lookAheadCoordinate).wrap(min: 0, max: 360),
                                   speed: currentSpeed,
                                   timestamp: Date())
-        
+
         self.simulatedLocation = location
-        
+
         delegate?.locationManager?(self, didUpdateLocations: [location])
```

Comment:
Should this call me made on the main queue?

Reply:
Probably yes.
https://github.com/mapbox/mapbox-navigation-ios/pull/3672#discussion_r833390975

Reply:
@bamx23 multiple tests and test fixtures rely on a synchronous delegate call inside the `tick` method. Since we're not guaranteeing execution on a specific queue, I'm proposing leaving it on a custom background queue.